staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -1067,21 +1121,58 @@ func emitCallsToValidators(c *generator.Context, validations []validators.Functi
 	}
 }
 
-func toGolangSourceDataLiteral(value any) string {
+func toGolangSourceDataLiteral(sw *generator.SnippetWriter, c *generator.Context, value any) {
 	// For safety, be strict in what values we output to visited source, and ensure strings
 	// are quoted.
-	switch value.(type) {
-	case uint, uint8, uint16, uint32, uint64, int8, int16, int32, int64, float32, float64, bool:
-		return fmt.Sprintf("%v", value)
+
+	switch v := value.(type) {
+	case uint, uint8, uint16, uint32, uint64, int, int8, int16, int32, int64, float32, float64, bool:
+		sw.Do(fmt.Sprintf("%v", value), nil)
 	case string:
 		// If the incoming string was quoted, we still do it ourselves, JIC.
 		str := value.(string)
 		if s, err := strconv.Unquote(str); err == nil {
 			str = s
 		}
-		return fmt.Sprintf("%q", str)
+		sw.Do(fmt.Sprintf("%q", str), nil)
+	case *types.Type:
+		sw.Do("$.|raw$", v)
+	case types.Member:
+		sw.Do("obj."+v.Name, nil)
+	case validators.PrivateVar:
+		sw.Do("$.|private$", c.Universe.Type(types.Name(v)))
+	case *validators.PrivateVar:
+		sw.Do("$.|private$", c.Universe.Type(types.Name(*v)))
+	case validators.Var:
+		sw.Do("$.|raw$", c.Universe.Type(types.Name(v)))
+	default:
+		rv := reflect.ValueOf(value)
+		switch rv.Kind() {
+		case reflect.Slice, reflect.Array:
+			arraySize := ""
+			if rv.Kind() == reflect.Array {
+				arraySize = strconv.Itoa(rv.Len())
+			}
+			var itemType string
+			switch rv.Type().Elem().Kind() {
+			case reflect.String: // For now, only support lists of strings.
+				itemType = rv.Type().Elem().Name()
+			default:
+				klog.Fatalf("Unsupported extraArg type: %T", value)
+			}
+			rv.Type().Elem()
+			sw.Do("[$.arraySize$]$.itemType${", map[string]string{"arraySize": arraySize, "itemType": itemType})
+			for i := 0; i < rv.Len(); i++ {
+				val := rv.Index(i)
+				toGolangSourceDataLiteral(sw, c, val.Interface())
+				if i < rv.Len()-1 {
+					sw.Do(", ", nil)
+				}
+			}
+			sw.Do("}", nil)
+		default:
+			// TODO: check this during discovery and emit an error with more useful information
+			klog.Fatalf("Unsupported extraArg type: %T", value)
```

Comment:
I thought I fixed all these, but this should be: `panic(fmt.Sprintf(...))`

Reply:
My bad. Fixing all occurrences I've introduced.